src/app/modules/payments/payments.service.ts

```diff
@@ -254,3 +255,41 @@ export const sendPaymentConfirmationEmailByPaymentId = (
       })
     })
 }
+
+/**
+ * Retrieves the latest payment document by email and formId.
+ * @param email the email of the payment to be retrieved
+ * @param formId the formId of the payment to be retrieved
+ * @returns ok(payment) if payment exists
+ * @returns err(PaymentNotFoundError) if the payment does not exist
+ * @returns err(DatabaseError) if error occurs whilst querying the database
+ */
+export const findLatestSuccessfulPaymentByEmailAndFormId = (
+  email: IPaymentSchema['email'],
+  formId: string,
+): ResultAsync<IPaymentSchema, PaymentNotFoundError | DatabaseError> => {
+  return ResultAsync.fromPromise(
+    PaymentModel.findOne({
+      email: email,
+      formId: formId,
+      status: PaymentStatus.Succeeded,
+    })
+      .sort({ _id: -1 })
+      .exec(),
+    (error) => {
+      logger.error({
+        message:
+          'Database error while finding latest payment by email and FormId',
+        meta: {
+          action: 'findLatestPaymentByEmailAndFormId',
+          email,
```

Comment:
Add `formId` in the meta block (to match our DD facet)

Reply:
Thanks for catching that! [4dc5c60](https://github.com/opengovsg/FormSG/pull/6170/commits/4dc5c601d0a6879130085ad9ee8f68365c6d0ac0)